java/java.source.base/src/org/netbeans/modules/java/source/save/Reformatter.java

```diff
@@ -2565,7 +2565,39 @@ public Boolean visitIf(final IfTree node, Void p) {
             StatementTree elseStat = node.getElseStatement();
             CodeStyle.BracesGenerationStyle redundantIfBraces = cs.redundantIfBraces();
             int eoln = findNewlineAfterStatement(node);
-            if ((elseStat != null && redundantIfBraces == CodeStyle.BracesGenerationStyle.ELIMINATE && danglingElseChecker.hasDanglingElse(node.getThenStatement())) ||
+            Boolean hasErrThenStatement = Optional.ofNullable(node.getThenStatement())
+                    .filter(it-> it instanceof ExpressionStatementTree)
+                    .map(it -> ((ExpressionStatementTree)it).getExpression())
+                    .map(it->  (it instanceof ErroneousTree))
+                    .orElse(false);
+            if (hasErrThenStatement) {
+                if (getCurrentPath().getParentPath().getLeaf() instanceof BlockTree parentStTree) {
+                    var isPreviousIfTree = false;
+                    var endPositionOfErrThenStatement = endPos;
+                    for (var statment : parentStTree.getStatements()) {
+                        if (isPreviousIfTree) {
+                            var startPositionOfNextErrorStatement = (int) sp.getStartPosition(getCurrentPath().getCompilationUnit(), statment);
```

Comment:
lets not use `var` here. `int` has 3 chars too - no need for extra obfuscation just to make the IDE draw the type as inline hint wasting space ;)

Makes it also easier to review without checking this out in NetBeans.

`var` is more interesting for collections with complex generics signatures etc.

Reply:
+1  Also `statment` -> `statement`.